### PR TITLE
Suppress empty summary

### PR DESCRIPTION
**Pull request by reedstrm**
Refs https://github.com/Connexions/webview/pull/1460

### CODE REVIEW

File: cnxepub/tests/test_formatters.py [hunk 1]
```diff
@@ -433,6 +433,10 @@ def test_binder(self):
             expected_content = f.read()
 
         actual = str(SingleHTMLFormatter(self.desserts))
+        # Uncomment this if/when you get a test failure
```
Comment (karenc) [inline cnxepub/tests/test_formatters.py]: Hm... Not sure if it's a good idea.  It's not that hard to add this if necessary is it?

File: cnxepub/tests/data/test.xhtml [hunk 2]
```diff
@@ -0,0 +1,314 @@
+<html xmlns="http://www.w3.org/1999/xhtml">
```
Comment (reedstrm) [inline cnxepub/tests/data/test.xhtml]: oops. Accidental `git add`. Will remove.

File: cnxepub/tests/test_formatters.py (hunk 1 above)
Comment (reedstrm) [inline cnxepub/tests/test_formatters.py]: This is the 4th time I've typed that from memory. Usually get it wrong at least once. :-/

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[flake8]
+exclude = 
+#    tests,
```
Comment (reedstrm) [inline .flake8]: yeah, handled by the next stanza down, where I use flake8-putty to ignore a subset of errors (mostly line length, and some style things about unused variables)

Comment (reedstrm) [inline .flake8]: Grr, I made the fixes Friday, even committed, but failed to push! :facepalm:

File: cnxepub/tests/test_formatters.py (hunk 1 above)
Comment (reedstrm) [inline cnxepub/tests/test_formatters.py]: Hmm, that works, but makes the test less efficient.  One of the other tests does do that, over in archive, I think). o.k. let me see what I can cook up. Ohh, can I make it conditional on the test passing? Then there's no commenting to do!

Comment (karenc) [inline cnxepub/tests/test_formatters.py]: Looks much better :smile: . Why not use a fixup commit?

File: cnxepub/tests/test_formatters.py [hunk 4]
```diff
@@ -433,7 +433,16 @@ def test_binder(self):
             expected_content = f.read()
 
         actual = str(SingleHTMLFormatter(self.desserts))
+        out_path = os.path.join(TEST_DATA_DIR,
+                                'desserts-single-page-actual.xhtml')
+        if not IS_PY3:
+            out_path = out_path.replace('.xhtml', '-py2.xhtml')
+
+        with open(out_path, 'w') as out:
+                out.write(actual)
```
Comment (pumazi) [inline cnxepub/tests/test_formatters.py]: wild indent here...

File: cnxepub/formatters.py [hunk 5]
```diff
@@ -604,12 +604,15 @@ def _fix_namespaces(html):
...
 
+      {% if metadata['summary'] %}
       <div class="description"
            itemprop="description"
            data-type="description"
            >
         {{ metadata['summary'] }}
       </div>
+      {%- endif %}
+
```
Comment (pumazi) [inline cnxepub/formatters.py]: extra line...

Comment (reedstrm) [inline cnxepub/formatters.py]: not extra I'm pretty sure the tests don't pass without it.

**Pull request merged**